protocol.go

```diff
@@ -49,13 +49,16 @@ type Protocol struct {
 	doc string
 
 	hash string
+
+	originalString string
 }
 
 // NewProtocol creates a protocol instance.
 func NewProtocol(
 	name, namepsace string,
 	types []NamedSchema,
 	messages map[string]*Message,
+	protocolString string,
```

Comment:
This is a breaking change. It cannot be added until the next major

Reply:
Let me create two different PRs. Or let me find a way to avoid this new parameter

Reply:
https://github.com/hamba/avro/pull/315#issuecomment-1772860065 added a comment here in the meantime

Reply:
Moving discussion in another PR https://github.com/hamba/avro/pull/316/files